staging/src/k8s.io/component-base/logs/example/cmd/logger.go

```diff
@@ -40,18 +51,25 @@ func NewLoggerCommand() *cobra.Command {
 	o := logs.NewOptions()
 	cmd := &cobra.Command{
 		Run: func(cmd *cobra.Command, args []string) {
-			if err := o.ValidateAndApply(); err != nil {
+			if err := o.ValidateAndApply(featureGate); err != nil {
 				fmt.Fprintf(os.Stderr, "%v\n", err)
 				os.Exit(1)
 			}
-			runLogger()
+
+			// Initialize contextual logging.
+			logger := klog.Background().WithName("example").WithValues("foo", "bar")
+			ctx := klog.NewContext(context.Background(), logger)
+
+			runLogger(ctx)
 		},
 	}
+	logs.AddFeatureGates(featureGate)
+	featureGate.AddFlag(cmd.Flags())
 	o.AddFlags(cmd.Flags())
 	return cmd
 }
 
-func runLogger() {
+func runLogger(ctx context.Context) {
```

Comment:
Should we also update the README that lists expected outputs for running example?
Adding a example on how to enable contextual logging would also be useful.

Reply:
That README indeed had to be updated. We need [automated checking of command lines](https://github.com/kubernetes/kubernetes/issues/102237)...

As the example update depends on feature gate support, I combined those two commits.